src/components/RadioGroup/RadioGroup.tsx

```diff
@@ -145,16 +148,16 @@ export const RadioGroup: ChakraComponent<
       );
 
       return (
-        <Fieldset
+        <Box
           className={className}
           id={`radio-group-${id}`}
-          isLegendHidden={!showLabel}
-          isRequired={isRequired}
-          legendText={labelText}
-          showRequiredLabel={showRequiredLabel}
           {...rest}
           __css={styles}
         >
+          <Box as="span" __css={styles.spanLegend}>
```

Comment:
For exact style parity, this element should have `display: block` applied. That is what was applied to the original `<legend>` element. Currently, this is using `display: inline-block`, which is essentially adding 2px padding to the top and creating a minor alignment issue.